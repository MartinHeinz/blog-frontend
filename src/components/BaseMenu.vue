<template>
        <span id="nav">
            <ul>
                <BaseMenuItem v-for="item in items"
                              v-bind:url="item.url"
                              v-bind:value="item.value"
                              v-bind:key="item.id">
                </BaseMenuItem>
            </ul>
        </span>
</template>

<script>

import BaseMenuItem from '@/components/BaseMenuItem.vue';

export default {
    name: 'BaseMenu',
    components: { BaseMenuItem },
    props: {
        items: Array,
    },
};
</script>

<style>

</style>
