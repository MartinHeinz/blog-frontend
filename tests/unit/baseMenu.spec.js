import { createLocalVue, mount } from '@vue/test-utils';
import BaseMenu from '@/components/BaseMenu.vue';
import Router from 'vue-router';

const items = [
    { value: 'Home', url: '/' },
    { value: 'Contact', url: '/contact' },
    { value: 'About', url: '/about' },
];

describe('BaseMenu.vue', () => {
    it('renders correcly', () => {
        const localVue = createLocalVue();
        const router = new Router({ mode: 'history' });
        localVue.use(Router);
        const wrapper = mount(BaseMenu, {
            localVue,
            router,
            propsData: {
                items,
            },
        });
        expect(wrapper.html()).toMatchSnapshot();
    });
});
