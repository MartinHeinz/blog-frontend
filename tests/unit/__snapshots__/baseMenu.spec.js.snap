// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseMenu.vue renders correcly 1`] = `<span id="nav"><ul><a href="/" class="router-link-exact-active router-link-active">Home</a><a href="/contact" class="">Contact</a><a href="/about" class="">About</a></ul></span>`;
